src/IO/Observer/Tags.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+
+/// \cond
+class DataVector;
+namespace observers {
+class ArrayComponentId;
+}  // namespace observers
+/// \endcond
+
+namespace observers {
+/// \ingroup ObserversGroup
+/// %Tags used on the observer parallel component
+namespace Tags {
+/// The number of events registered with the observer.
+struct NumberOfEvents : db::SimpleTag {
+  using type =
+      std::unordered_map<ArrayComponentId, std::unique_ptr<std::atomic_int>>;
+  static std::string name() noexcept { return "NumberOfEvents"; }
+};
+
+/// All the ids of all the components registered to an observer for doing
+/// reduction observations.
+struct ReductionArrayComponentIds : db::SimpleTag {
+  using type = std::unordered_set<ArrayComponentId>;
+  static std::string name() noexcept { return "ReductionArrayComponentIds"; }
+};
+
+/// All the ids of all the components registered to an observer for doing
+/// volume observations.
+struct VolumeArrayComponentIds : db::SimpleTag {
+  using type = std::unordered_set<ArrayComponentId>;
+  static std::string name() noexcept { return "VolumeArrayComponentIds"; }
+};
+
+/// Volume tensor data to be written to disk.
+struct TensorData : db::SimpleTag {
```

Comment:
It would be helpful to explain what is expected to be stored in something tagged by this (i.e. its relation to `type`)

Reply:
Yea... Not yet completely clear :P I'm going to expand on this in the next PR if that's alright? I'm not completely certain that this won't evolve a bit yet.